FeedStoreChallenge/RealmFeedStore.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  RealmFeedStore.swift
+//  FeedStoreChallenge
+//
+//  Created by Thiago Penna on 09/02/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import RealmSwift
+
+public class RealmFeedStore: FeedStore {
+	private let configuration: Realm.Configuration
+	private var realm: Realm?
+	
+	public init(configuration: Realm.Configuration) {
+		self.configuration = configuration
+	}
```

Comment:
The configuration is only used to create the `realm` instance once. So it could be done here in the initializer. This way, you won't have an optional `realm` property. Something like:

```suggestion
	private let realm: Realm
	
	public init(configuration: Realm.Configuration) throws {
		self.realm = try RealmFeedStore.createRealmInstance(with: configuration)
	}
```

Reply:
I went for exactly this suggestion with a throwable initializer. The only problem is now I can't find a way to make the insertion or deletion operations to fail (since earlier I was depending on the Realm instance creation to fail for those tests). But keeping it this way really simplifies the production implementation, while also solving any problems regarding in-memory realms (since the RealmFeedStore will keep a reference to it during its lifetime).